.github/workflows/rust_binary_publish.yml

```diff
@@ -155,6 +156,22 @@ jobs:
           sas=$(az storage blob generate-sas --connection-string "${{ secrets.ARTIFACTS_CONNECTION_STRING }}" --container-name ${{ vars.ARTIFACTS_CONTAINER }} --name ${{  inputs.package }}/${{ inputs.release_channel }}/${{  inputs.package }}-${{ matrix.target }}-${{  inputs.toolchain}}-v${{  inputs.version }}${{ fromJson(inputs.targets_config)[matrix.target].extension }} --permissions r --expiry ${{ steps.sas-expiry.outputs.expiry }} --https-only --full-uri --output tsv)
           echo "sas=$sas" >> $GITHUB_OUTPUT
 
+      - name: Generate SAS Url for binary signing upload
+        if: inputs.sign_build == 'true'
+        shell: bash
+        id: sas-upload-url
+        run: |
+          sas=$(az storage blob generate-sas --connection-string "${{ secrets.ARTIFACTS_CONNECTION_STRING }}" --container-name ${{ vars.ARTIFACTS_CONTAINER }} --name ${{  inputs.package }}/${{ inputs.release_channel }}/${{  inputs.package }}-${{ matrix.target }}-${{ inputs.toolchain }}-v${{ inputs.version }}-signed${{ fromJson(inputs.targets_config)[matrix.target].extension }} --permissions w --expiry ${{ steps.sas-expiry.outputs.expiry }} --https-only --full-uri --output tsv)
```

Comment:
Might want to put double quotes around the blob name (--name) if it can contain spaces